src/UndoCommands.cpp

```diff
@@ -292,18 +360,35 @@ redo()
 {
   _scene->clearSelection();
 
-  _newSceneJson = takeSceneJsonFromClipboard();
-
-  if (_newSceneJson.empty())
-    return;
-
-  _newSceneJson = makeNewNodeIdsInScene(_newSceneJson);
+  // Ignore if pasted in content that does not generate nodes.
+  try
+  {
+    insertSerializedItems(_newSceneJson, _scene);
+  }
+  catch(...)
+  {
+    auto & graphModel = _scene->graphModel();
 
-  QPointF averagePos = computeAverageNodePosition(_newSceneJson);
+    QJsonArray connJsonArray;
+    for (QGraphicsItem * item : _scene->selectedItems())
+    {
+      if (auto c = qgraphicsitem_cast<ConnectionGraphicsObject*>(item))
+      {
+        graphModel.deleteConnection(c->connectionId());
+      }
+    }
 
-  offsetNodeGroup(_newSceneJson, _mouseScenePos - averagePos);
+    QJsonArray nodesJsonArray;
+    for (QGraphicsItem * item : _scene->selectedItems())
+    {
+      if (auto n = qgraphicsitem_cast<NodeGraphicsObject*>(item))
+      {
+        graphModel.deleteNode(n->nodeId());
```

Comment:
I think `setSelected` is missing for the created connections in the function `insertSerializedItems`

Reply:
Yes, I have amended